### ISSUE TITLE
iOS - Unable to download/play local files

### ISSUE THREAD

**Issue by slowspike** [advplyr/audiobookshelf-app#848]
### Steps to reproduce

1. Download any audiobook (`.mp3` or `.m4b`)
2. Try to play local files
3. Receive error "Playback Failed: Playback Error"

### Expected behaviour
- Tell us what should happen
File should download and be playable.

### Actual behaviour
- Tell us what happens
File seems to download too fast for the file size and does not play. Streaming the file works on the iOS app. Using the app on Android, the download is successful and local playback works. 

### Environment data

Audiobookshelf Version: app v0.9.65-beta/server v2.3.3

- [ ] Android App?
- [X] iOS App?

#### iOS Issue
iOS Version: 16.6

iPhone model: iPhone 13 Pro

**Comment by nichwall** [comment]
Is this a new install?

There were some reports of issues like this after updating the server, and resetting the iOS app fixed the issues.

**Comment by slowspike** [comment]
Yes, both server and apps are new installs. 

I tried restarting the iphone and uninstalling/reinstalling app, but downloads still aren't working on ios. With the android app, I get a confirmation box asking if I wanted to download and also see the download in the server logs.

**Comment by advplyr** [comment]
What is the file format? It can also be helpful to use the web UI to get the full probe data

**Comment by slowspike** [comment]
I've tried both `.m4b` and `.mp3` on ios. I just tested downloading using the web UI and that works as well.

**Comment by advplyr** [comment]
Are you using a reverse proxy or downloading from local ip?

**Comment by slowspike** [comment]
I have been using a reverse proxy (nginx proxy manager) and have websockets support on. I just tried using a local ip in the app instead and the download works as intended. Any suggestions for settings to tweak?

**Comment by realmain** [comment]
Are you using Cloudflare as well? Because Cloudflare only allows up to 100MB uploads on the free plan.

**Comment by slowspike** [comment]
I do use Cloudflare, but I am able to download the same files using the android app.

**Comment by realmain** [comment]
That is strange. For me, it's impossible to download any audiobooks on my 4 Android devices if I connect via my website that routes through the Cloudflare proxy due to the 100MB limit, which is understandable. I have to always use a direct connection (or I could untoggle my Cloudflare proxy).

Maybe your Android device is bypassing it somehow, because a 100MB limit is a 100MB limit.

**Comment by slowspike** [comment]
I just tried downloading through the webui using the reverse proxy and it works as well. 

My understanding is the 100MB limit is per request and chunking uploads is a way to share larger files. For example, I'm able to upload and share files much larger than 100MB with nextcloud using a reverse proxy, but I did have to change some settings in nextcloud, like `client_max_body_size`.

Are you able to download through the webui when using the reverse proxy?

**Comment by realmain** [comment]
Disregard my comments... Yes, downloading via the web UI works, and also tested 3 of my android devices and the app works now. Was not working a couple months ago and I didn't try it again after updates. I just assumed it was Cloudflare.

**Comment by slowspike** [comment]
I'm hoping that downloads in iOS (in my particular case) will be ironed out in later versions of the app.

**Comment by anddam** [comment]
Seeing same issue with server 2.3.4 and iOS 0.9.66-beta, new install on the iPhone.

**Comment by advplyr** [comment]
I'm not able to reproduce this using nginx proxy manager and cloudflare

**Comment by anddam** [comment]
> I'm not able to reproduce this using nginx proxy manager and cloudflare

Oh I am not even talking about public networknig, I have the audiobookshelf server in LAN, the clients, both Android and iOS, are connecting straight away via http. I do see the download progressing on iOS, as I do on Android, yet the playback fails as in the issue description.

As additional detail streaming correctly works on both platforms.

**Comment by slowspike** [comment]
I was able to fix my issue. With server 2.3.4 and iOS 0.9.66-beta, I wasn't even able to login anymore using a reverse proxy on the iOS app. On my Cloudflare dashboard, I saw under Security>Events that one of my custom WAF rules (block bad bots) was blocking the app from logging in. I whitelisted the ASN under IP Access Rules, and was able to login. Finally, when I tried to download an audiobook, it was successful and could be played. 

My original problem was not being able to download, but I didn't check my Cloudflare settings at the time, so I'm not sure if a WAF was blocking just the downloads then.
